dask/array/reductions.py

```diff
@@ -423,12 +428,20 @@ def min(a, axis=None, keepdims=False, split_every=None, out=None):
     )
 
 
+def chunk_min(x, axis=None, keepdims=None, *args, **kwargs):
+    if x.size == 0:
+        return np.array([sys.maxsize])
```

Comment:
Instead let's return an empty array here (still with the correct `dtype`). This way `chunk_min` will avoid the `ValueError: zero-size array to reduction operation maximum which has no identity` error. We can then use `combine=chunk.min` to ensure that if _all_ chunks are empty, we still match NumPy's behavior and raise that zero-size error. 

Specifically, changes along the lines of: 

```diff
diff --git a/dask/array/reductions.py b/dask/array/reductions.py
index 97b7f2221..a54006d11 100644
--- a/dask/array/reductions.py
+++ b/dask/array/reductions.py
@@ -419,7 +419,8 @@ def min(a, axis=None, keepdims=False, split_every=None, out=None):
     return reduction(
         a,
         chunk_min,
-        chunk.min,
+        chunk_min,
+        combine=chunk.min,
         axis=axis,
         keepdims=keepdims,
         dtype=a.dtype,
@@ -429,8 +430,9 @@ def min(a, axis=None, keepdims=False, split_every=None, out=None):


 def chunk_min(x, axis=None, keepdims=None, *args, **kwargs):
+    """Version of np.min which ignores size 0 arrays"""
     if x.size == 0:
-        return np.array([sys.maxsize])
+        return np.array([], dtype=x.dtype)
     else:
         r = np.min(x, axis=axis, keepdims=keepdims, *args, **kwargs)
         return r
@@ -442,7 +444,8 @@ def max(a, axis=None, keepdims=False, split_every=None, out=None):
     return reduction(
         a,
         chunk_max,
-        chunk.max,
+        chunk_max,
+        combine=chunk.max,
         axis=axis,
         keepdims=keepdims,
         dtype=a.dtype,
@@ -452,8 +455,9 @@ def max(a, axis=None, keepdims=False, split_every=None, out=None):


 def chunk_max(x, axis=None, keepdims=None, *args, **kwargs):
+    """Version of np.max which ignores size 0 arrays"""
     if x.size == 0:
-        return np.array([-sys.maxsize])
+        return np.array([], dtype=x.dtype)
     else:
         r = np.max(x, axis=axis, keepdims=keepdims, *args, **kwargs)
         return r
```

should work

Reply:
I implemented this and tried computing the maximum of `a = da.arange(5, chunks=2)` and it works. However, if I change the chunk size to `chunks=1`, it gives `ValueError: zero-size array to reduction operation maximum which has no identity`.

Reply:
Hrm, I don't seem to be able to reproduce that locally with these suggestion changes:

```python
In [1]: import dask.array as da

In [2]: a = da.arange(5, chunks=2)

In [3]: da.max(a)
Out[3]: dask.array<chunk_max-aggregate, shape=(), dtype=int64, chunksize=(), chunktype=numpy.ndarray>

In [4]: da.max(a).compute()
Out[4]: 4

In [5]: a = da.arange(5, chunks=1)

In [6]: a
Out[6]: dask.array<arange, shape=(5,), dtype=int64, chunksize=(1,), chunktype=numpy.ndarray>

In [7]: da.max(a).compute()
Out[7]: 4
```

Are you able to?

Reply:
Sorry, I meant to say computing `a[a<2]` maximum

Reply:
Ah, I see. It looks like I:

- Got my `combine` and `aggregate` functions mixed up
- Didn't account for arrays with `ndim > 1` (which is why `test_searchsorted` was failing)

Locally it looks like

```python
@implements(np.max, np.amax)
@derived_from(np)
def max(a, axis=None, keepdims=False, split_every=None, out=None):
    return reduction(
        a,
        chunk_max,
        chunk.max,
        combine=chunk_max,
        axis=axis,
        keepdims=keepdims,
        dtype=a.dtype,
        split_every=split_every,
        out=out,
    )


def chunk_max(x, axis=None, keepdims=None, *args, **kwargs):
    if x.size == 0:
        return np.array([], ndmin=x.ndim, dtype=x.dtype)
    else:
        r = np.max(x, axis=axis, keepdims=keepdims, *args, **kwargs)
        return r
```

works for me. Note switching the `aggregate` and `combine` functions in the `reduction` call and adding `ndmin=` in `chunk_max`